test/sslapitest.c

```diff
@@ -9831,6 +9876,255 @@ static int test_load_dhfile(void)
+                                       TLS_client_method(),
+                                       TLS1_3_VERSION, 0,
+                                       &sctx, &cctx, cert, privkey)))
+        goto end;
+    if (!TEST_true(SSL_CTX_set_cert_comp_preference(sctx, server_pref, 3)))
+        goto end;
+    if (!TEST_true(SSL_CTX_set_cert_comp_preference(cctx, client_pref, 3)))
+        goto end;
+
+    switch (test) {
+    case 0: /* don't do cert compression */
+        if (!TEST_true(SSL_CTX_set_cert_comp_cb(sctx, cert_comp_cb0, &result_server)))
+            goto end;
+        if (!TEST_true(SSL_CTX_set_cert_comp_cb(cctx, cert_comp_cb0, &result_client)))
+            goto end;
+        break;
+    case 1: /* have libssl do cert compression */
+        if (!TEST_true(SSL_CTX_set_cert_comp_cb(sctx, cert_comp_cb1, &result_server)))
+            goto end;
+        if (!TEST_true(SSL_CTX_set_cert_comp_cb(cctx, cert_comp_cb1, &result_client)))
+            goto end;
+        break;
+    case 2: /* error while doing cert compression */
+        if (!TEST_true(SSL_CTX_set_cert_comp_cb(sctx, cert_comp_cb2, &result_server)))
+            goto end;
+        if (!TEST_true(SSL_CTX_set_cert_comp_cb(cctx, cert_comp_cb2, &result_client)))
+            goto end;
+        break;
+    case 3: /* have libssl do cert compression, client auth */
+        if (!TEST_true(SSL_CTX_set_cert_comp_cb(sctx, cert_comp_cb1, &result_server)))
+            goto end;
+        if (!TEST_true(SSL_CTX_set_cert_comp_cb(cctx, cert_comp_cb1, &result_client)))
+            goto end;
+        /* Use callbacks from test_client_cert_cb() */
+        SSL_CTX_set_client_cert_cb(cctx, client_cert_cb);
+        SSL_CTX_set_verify(sctx, SSL_VERIFY_PEER | SSL_VERIFY_FAIL_IF_NO_PEER_CERT, verify_cb);
+        break;
+    case 4: /* application does the compression */
+        if (!TEST_true(SSL_CTX_set_cert_comp_cb(sctx, cert_comp_cb4, &result_server)))
+            goto end;
+        if (!TEST_true(SSL_CTX_set_cert_comp_cb(cctx, cert_comp_cb4, &result_client)))
+            goto end;
+        break;
+    default:
+        break;
+    }
+
+    if (!TEST_true(create_ssl_objects(sctx, cctx, &serverssl, &clientssl,
+                                      NULL, NULL)))
+        goto end;
+
+    if (!TEST_true(SSL_set_app_data(clientssl, &client_seen)))
+        goto end;
+    if (!TEST_true(SSL_set_app_data(serverssl, &server_seen)))
+        goto end;
+    if (test == 0) {
+        SSL_set_info_callback(clientssl, no_cert_comp_info_cb);
+        SSL_set_info_callback(serverssl, no_cert_comp_info_cb);
+    } else {
+        SSL_set_info_callback(clientssl, cert_comp_info_cb);
+        SSL_set_info_callback(serverssl, cert_comp_info_cb);
+    }
+
+    /* This is the error-out test... */
+    if (test == 2) {
+        if (!TEST_false(create_ssl_connection(serverssl, clientssl, SSL_ERROR_NONE)))
+            goto end;
+        goto pass;
```

Comment:
Just an `else` would be neater

Reply:
`else` for the `goto pass;` or `else` for the remaining `TEST`s?

Reply:
I mean drop the `goto pass` in favour of an `else`